src/main/java/org/opensearch/knn/index/MethodComponentContext.java

```diff
@@ -193,6 +197,69 @@ public Map<String, Object> getParameters() {
         return parameters;
     }
 
+    @Override
+    public String toString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("{name=").append(name).append(";");
+        stringBuilder.append("parameters=[");
+        if (Objects.nonNull(parameters)) {
+            for (Map.Entry<String, Object> entry : parameters.entrySet()) {
+                stringBuilder.append(entry.getKey()).append("=");
```

Comment:
Can you add a comment that these will always be either strings, ints, or methodcomponentcontexts and reference the parameter.java file?